annubes/tasks/task.py

```diff
@@ -1,209 +1,211 @@
-            trials (dict): dictionary containing all the trials generated and the task's relevant information.
-            modalities (list[str]): list of individual modalities.
-            modality_idx (dict[str, int]): dictionary of modalities and their index.
-            n_in (int): number of inputs.
-            imin (float): minimum value of the input intensities.
-            imax (float): maximum value of the input intensities.
-            coeff (float): coefficient for scaling the input.
-            T (int): total time of each trial in ms.
-            t (np.ndarray): time vector of each trial in ms.
-            low_out (float): low intensity value for the output signals.
-            high_out (float): high intensity value for the output signals.
-
-        Raises:
-            ValueError: if the sum of the probabilities of `session_in` is not 1.
-            ValueError: if `catch_prob` is not between 0 (included) and 1 (excluded).
-        """
-        # Attributes from the class constructor parameters
-        self.name = name
-        self.session_in = OrderedDict(session_in) if session_in else OrderedDict({"v": 0.5, "a": 0.5})
-        self.ordered = ordered
-        self.t_in = t_in
-        self.value_in = value_in if value_in else [0.8, 0.9, 1]
-        self.scaling = scaling
-        self.t_fixation = t_fixation
-        self.value_fixation = value_fixation
-        self.max_sequential = max_sequential
-        self.catch_prob = catch_prob
-        self.inter_trial = inter_trial
-        self.dt = dt
-        self.tau = tau
-        self.std_inp_noise = std_inp_noise
-        self.baseline_inp = baseline_inp
-        self.n_out = n_out
-        self.value_out = value_out if value_out else [0, 1]
+    """General class for defining a task.
+
+    Args:
+        name: Name of the task.
+        session: Dictionary representing the ratio (values) of the different trials (keys) within the task. Trials with
+            a single modality must be represented by single characters, while trials with multiple modalities are
+            represented by the character combination of those trials. Note that values are read relative to each other,
+            such that e.g. `{"v": 0.25, "a": 0.75}` == `{"v": 1, "a": 3}` is True. Defaults to {"v": 0.5, "a": 0.5}.
+        stim_intensities: List of possible intensities of each stimulus.
+            Defaults to [0.8, 0.9, 1].
+        catch_prob: probability of catch trials in the session, between 0 and 1.
+            Defaults to 0.5.
+        catch_intensity: Intensity value during a catch trial.
+            Defaults to 0.
+        shuffle_trials: If True (default), trial order will be randomized. If False, all trials of one modality are run
+            before any trial of the next modality starts, in the order defined in `session` followed by catch trials.
+            Note that in the latter case, `self.session` will be converted to an `OrderedDict`, such that e.g.,
```

Comment:
I'd remove the sentence from "Note that [...]", since to me it sounds like too much detail for a doc string meant more for users than for developers.